src/main/java/uk/gov/register/store/postgres/PostgresReadDataAccessLayer.java

```diff
@@ -0,0 +1,168 @@
+package uk.gov.register.store.postgres;
+
+import uk.gov.register.core.Entry;
+import uk.gov.register.core.Item;
+import uk.gov.register.core.Record;
+import uk.gov.register.db.*;
+import uk.gov.register.store.DataAccessLayer;
+import uk.gov.register.util.HashValue;
+
+import java.time.Instant;
+import java.util.*;
+import java.util.function.Function;
+
+public abstract class PostgresReadDataAccessLayer implements DataAccessLayer {
```

Comment:
As discussed I kept these separate just to make it easier to visually see where the batching logic is and the straight-to-database logic is. In the work coming up I think these two classes will get a lot smaller and we should be able to combine them or split them in a more sensible way.